app.py

```diff
@@ -32,6 +31,45 @@ def wrapper(*args, **kw):
     return wrapper
 
 
+def recover_identity(f):
+    @wraps(f)
+    def wrapper(*args, **kw):
+        # Authorization request header format:
+        # Authorization: Signature <signature_base64_encoded>
+        authorization = request.headers.get('Authorization')
+        if not authorization:
+            return jsonify(error='missing Authorization in request header'), 401
+
+        if authorization.count(' ') != 1:
+            return jsonify(error='invalid Authorization header value provided, correct format: Signature <signature_base64_encoded>'), 401
+
+        authentication_type, signature_base64_encoded = authorization.split(' ')
+
+        if authentication_type != 'Signature':
+            return jsonify(error='authentication type have to be Signature'), 401
+
+        if signature_base64_encoded == '':
+            return jsonify(error='signature was not provided'), 401
+
+        try:
+            signature_bytes = base64.b64decode(signature_base64_encoded)
+        except Exception as err:
```

Comment:
> A TypeError is raised if s is incorrectly padded.
I think it's safe to expect `TypeError` exception.
Catching general `Exception` is dangerous - a lot of thinks can cause exceptions, like keyboard interrupt (when you press ctrl+d), memory/os issues etc.

Reply:
agree. updated